lib/Request.js

```diff
@@ -276,7 +269,14 @@ class Request {
       } else {
 
         const outputBucket = this.target.bucket
-        const outputKey = key.replace(prefix, this.target.prefix)
+        let outputKey = key.replace(prefix, this.target.prefix)
+
+        // Rename if rename function exists
+        if (this.target.rename) {
+          outputKey = this.target.rename(outputKey).then(() => {
```

Comment:
we should run `rename` on the file name that comes after `prefix` instead of the whole key itself, since the prefix is already part of the output context